src/simulator/simulator.cpp

```diff
@@ -126,48 +139,106 @@ void Simulator::init(ros::NodeHandle& n) {
   odometryTwistMsg.header.frame_id = "odom";
   odometryTwistMsg.child_frame_id = "base_footprint";
 
-  cur_loc_ = Pose2Df(CONFIG_start_angle, {CONFIG_start_x, CONFIG_start_y});
-
-  // Create motion model based on robot type
-  // TODO extend to handle the multi-robot case
-  if (robot_type_ == "F1TEN") {
-    motion_model_ =
-          unique_ptr<AckermannModel>(new AckermannModel({CONFIG_robot_config}, &n));
-  } else if (robot_type_ == "COBOT") {
-    motion_model_ =
-          unique_ptr<CobotModel>(new CobotModel({CONFIG_robot_config}, &n));
-  } else if (robot_type_ == "BWIBOT") {
-    motion_model_ =
-          unique_ptr<DiffDriveModel>(new DiffDriveModel({CONFIG_robot_config}, &n));
+  for (int i = 0; i < robot_number_; i++) {
+    string robot_type = robot_types_[i];
+    string topic_prefix = topic_prefixs_[i];
+    if (robot_type == "F1TEN") {
+      motion_models_.push_back(unique_ptr<AckermannModel>(
+          new AckermannModel({CONFIG_robot_configs[i]}, &n, topic_prefix)));
+    } else if (robot_type == "COBOT"){
+      motion_models_.push_back(unique_ptr<CobotModel>(
+          new CobotModel({CONFIG_robot_configs[i]}, &n, topic_prefix)));
+    }
+    else if (robot_type == "BWIBOT") {
+      if(robot_number_ != 1){
+        printf("Warning, Diff Drive Currently does not support multiple robots\n");
+      }
+      motion_models_.push_back(unique_ptr<DiffDriveModel>(
+          new DiffDriveModel({CONFIG_robot_configs[i]}, &n)));
+    }
+    // initialize starting location
+    cur_locs_.push_back(
+        Pose2Df(CONFIG_robot_start_angles[i], CONFIG_robot_start_locations[i]));
+    motion_models_[i]->SetPose(cur_locs_[i]);
   }
-  motion_model_->SetPose(cur_loc_);
-  initSimulatorVizMarkers();
+
+  initSimulatorVizMarkers(); // TODO: need to add a vis marker for every robot
```

Comment:
add name in TODO

Reply:
Sorry thats already done. I will proceed with removing the comment.